### PR TITLE
Fix sem timeouts and priorities

### PR DESCRIPTION
**Pull request by Brian-McM**

### CODE REVIEW

**Review by bcreane — APPROVED** [review]
/lgtm

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and delete the branch after I've merged it.

**Comment by marvin-tigera** [comment]
I'm sorry but I failed delete the branch after merging the pull request.

**Pull request merged**
